tools/database/dbc/spell_effect.go

```diff
@@ -206,21 +200,45 @@ func (effect *SpellEffect) IsPeriodicDamageEffect() bool {
 func (data *SpellEffect) ClassFlag(index uint) uint32 {
 	return uint32(data.EffectSpellClassMasks[index/32]) & (1 << (index % 32))
 }
-
-func (effect *SpellEffect) ParseStatEffect() *stats.Stats {
-	stats := &stats.Stats{}
+func (effect *SpellEffect) CalcCoefficientStatValue(ilvl int) float64 {
+	propPoints := effect.GetScalingValue(ilvl)
+	return math.Round(float64(propPoints) * effect.Coefficient)
+}
+func (effect *SpellEffect) GetScalingValue(ilvl int) float64 {
+	if ilvl > 0 {
+		// If item we get rand prop points
+		return float64(dbcInstance.RandomPropertiesByIlvl[ilvl][proto.ItemQuality_ItemQualityEpic][0])
+	}
+	spell := dbcInstance.Spells[effect.SpellID]
+	// if not we get class scaling based on the spell
 	scale := effect.ScalingClass()
-	spell := dbcInstance.Spells[effect.ID]
+	return dbcInstance.SpellScalings[core.TernaryInt(spell.MaxScalingLevel > BASE_LEVEL, BASE_LEVEL, spell.MaxScalingLevel)].Values[scale]
```

Comment:
`core.TernaryInt(spell.MaxScalingLevel > BASE_LEVEL, BASE_LEVEL, spell.MaxScalingLevel)` can be simplified to just `min(spell.MaxScalingLevel, BASE_LEVEL)`.